src/cluster_legacy.c

```diff
@@ -4439,6 +4439,12 @@ void clusterLogCantFailover(int reason) {
         time(NULL) - lastlog_time < CLUSTER_CANT_FAILOVER_RELOG_PERIOD)
         return;
 
+    /* If data age is too old, this log may be printed repeatedly since it
+     * can not be automatically recovered. In this case, limit its frequency. */
+    if (reason == server.cluster->cant_failover_reason && reason == CLUSTER_CANT_FAILOVER_DATA_AGE &&
+        time(NULL) - lastlog_time < 10 * CLUSTER_CANT_FAILOVER_RELOG_PERIOD)
+        return;
+
```

Comment:
This seems a bit more readable? What do you think?

```suggestion
    time_t now = time(NULL);
    
    /* General logging suppression if the same reason has occurred recently. */
    if (now - lastlog_time < CLUSTER_CANT_FAILOVER_RELOG_PERIOD) {
        return;
    }

    /* Special case: If the failure reason is due to data age, log 10 times less frequently. */
    if (reason == CLUSTER_CANT_FAILOVER_DATA_AGE &&
        now - lastlog_time < 10 * CLUSTER_CANT_FAILOVER_RELOG_PERIOD) {
        return;
    }
}
```

Reply:
LGTM, thanks

Reply:
yeah I assumed there is a check of `reason == server.cluster->cant_failover_reason` before my suggestions :). github doesn't allow me to include that line in the suggested change ...